### PR TITLE
feat(golangci-lint): Add GOMEMLIMIT to env

### PR DESCRIPTION
**Pull request by Ned-el-ch**
Please read [CONTRIBUTING.md](CONTRIBUTING.md) for additional information on contributing to this repository!

## What this PR does / why we need it
Adds GOMEMLIMIT with a default of 6GiB to the e2e env variables

## Jira ID

[DT-4162]

## Notes for your reviewers

[DT-4162]: https://outreach-io.atlassian.net/browse/DT-4162?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by pbr0ck3r — APPROVED** [review]

**Comment by malept** [comment]
> We may want to document somewhere the different CirceCI image containers and what their max memory limit is. If a team increases container size to say xlarge to speed things up, they will want to bump the `gomemlimit` to take advantage.

Should this be part of the description?

**Review by yj-yan — APPROVED** [review]

**Review by malept — CHANGES_REQUESTED** [review]
I'm confused. The original issue was in the `shared/test` job, so why is the fix in the `shared/e2e` job?

**Comment by Ned-el-ch** [comment]
> I'm confused. The original issue was in the `shared/test` job, so why is the fix in the `shared/e2e` job?

the jira ticket said e2e so it went there but i'll change it 
> We could also also the GOMEMLIMIT in .circleci/config in the tests (e2e) steps, as it can't be set in a workflow via stencil-circleci. [Set an environment variable - CircleCI](https://circleci.com/docs/set-environment-variable/)

File: orbs/shared/jobs/test.yaml [hunk 1]
```diff
@@ -58,6 +62,7 @@ environment:
   AWS_REGION: << parameters.aws_region >>
   PRE_SETUP_SCRIPT: << parameters.pre_setup_script >>
   GO_TEST_TIMEOUT: << parameters.go_test_timeout >>
+  GOMEMLIMIT: << parameters.gomemlimit >>
```
Comment (malept) [inline orbs/shared/jobs/test.yaml:65]: issue: when you set this, the entire check for heuristically determining the value of `GOMEMLIMIT` is always bypassed due to this check:

https://github.com/getoutreach/devbase/blob/239fd4372b6e3d360cb351c000e96958b66025ef/shell/golangci-lint.sh#L44

We need to figure out a better way of allowing repos to specify the value without removing the heuristic calculations feature.

File: shell/golangci-lint.sh [hunk 2]
```diff
@@ -53,7 +53,9 @@ if [[ -z $GOGC ]] && [[ -z $GOMEMLIMIT ]]; then
   # This is mostly important for CI systems or other container
   # environments.
   if (command -v free || command -v sysctl) &>/dev/null; then
-    if command -v free &>/dev/null; then
+    if [ -f /sys/fs/cgroup/memory/memory.limit_in_bytes ]; then
+      mem=$(($(cat /sys/fs/cgroup/memory/memory.limit_in_bytes) / 1024 / 1024))
```
Comment (malept) [inline shell/golangci-lint.sh:57]: suggestion: it occurred to me that this doesn't seem to be related to `free` or `sysctl` being available, so this `if` statement could be on the same level as line 55, instead of being nested inside of line 55.
